sketch_loader.go

```diff
@@ -123,7 +123,9 @@ func makeSketch(sketchLocation string, allSketchFilePaths []string, logger i18n.
 				otherSketchFiles = append(otherSketchFiles, sketchFile)
 			}
 		} else if ADDITIONAL_FILE_VALID_EXTENSIONS[ext] {
-			additionalFiles = append(additionalFiles, sketchFile)
+			if !strings.Contains(filepath.Dir(sketchFile.Name), buildLocation) {
+				additionalFiles = append(additionalFiles, sketchFile)
+			}
```

Comment:
I believe this is not necessary, as recursive compilation is already limited to the `src/` subdirectory. Have you confirmed that this check ever returns false with a build dir inside the sketch dir?

Reply:
I checked and if I only remove the other line all temporary cpp files get added top the list (even if not in src)

Reply:
There's also a test that should not pass if files outside `src` were not added to the list (https://github.com/arduino/arduino-builder/blob/master/test/sketch_loader_test.go#L129) . Is it intended @cmaglie ?

Reply:
If the test is there it probably means something. you should go back to the commit that added those lines and check why they have been added...

Reply:
This is the commit that added the "extra" files to the test: 9d7d1d7cb95c12e73e9dd44df132a643089c3f40